test/framework/toolchain.py

```diff
@@ -816,6 +820,48 @@ def test_pgi_imkl(self):
         liblapack += "-Wl,--end-group -Wl,-Bdynamic -ldl"
         self.assertEqual(os.environ.get('LIBLAPACK', '(not set)'), liblapack)
 
+    def test_ccache(self):
+        """Test ccache"""
+        # generate shell script to mock ccache, f90cache
+        for cachename in ['ccache', 'f90cache']:
+            path = os.path.join(self.test_prefix, '%s' % cachename)
+
+            txt = [
+                "#!/bin/bash",
+                "echo 'This is a ccache wrapper'",
+                "NAME=${0##*/}",
+                "comm=$(which -a $NAME | sed 1d)",
+                "$comm $@",
+                "exit 0"
+            ]
+            script = '\n'.join(txt)
+            fn = os.path.join(path, '%s' % cachename)
+            write_file(fn, script)
+
+            # make script executable
+            st = os.stat(fn)
+            os.chmod(fn, st.st_mode | stat.S_IEXEC)
+            setvar('PATH', '%s:%s' % (path, os.getenv('PATH')))
+
+
+        eb_file = os.path.join(os.path.dirname(__file__), 'easyconfigs', 'toy-0.0.eb')
+
+        args = [
+            eb_file,
+            "--use-compiler-cache",
+            self.test_prefix,
```

Comment:
maybe change this to `"--use-compiler-cache=%s" % self.test_prefix` which makes it more explicit that `--use-compiler-cache` takes an argument